modules/terraform-azure-bastion/main.tf

```diff
@@ -0,0 +1,119 @@
+# Copyright 2025 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "azurerm_network_security_group" "bastion_sg" {
+  name                = var.bastion_security_group_name
+  location            = var.bastion_region
+  resource_group_name = var.bastion_resourcegroup_name
+}
+
+resource "azurerm_network_security_rule" "ssh" {
```

Comment:
Also, it might be worth having SG creation an optional step, similar to the `create_bastion_sg` input variable in the AWS bastion module (https://github.com/cloudera-labs/terraform-cdp-modules/blob/main/modules/terraform-aws-bastion/variables.tf#L30-L35)